lib/synapse/service_watcher/base.rb

```diff
@@ -33,16 +33,10 @@ def initialize(opts={}, synapse)
       @leader_election = opts['leader_election'] || false
       @leader_last_warn = Time.now - LEADER_WARN_INTERVAL
 
-      # the haproxy config
-      @haproxy = opts['haproxy']
-      @haproxy['server_options'] ||= ""
-      @haproxy['server_port_override'] ||= nil
-      %w{backend frontend listen}.each do |sec|
-        @haproxy[sec] ||= []
-      end
-
-      unless @haproxy.include?('port')
-        log.warn "synapse: service #{name}: haproxy config does not include a port; only backend sections for the service will be created; you must move traffic there manually using configuration in `extra_sections`"
+      @generator_config = {}
+      @synapse.available_generators.each do |generator_name, generator|
+        @generator_config[generator_name] = opts[generator_name] || {}
```

Comment:
i would prefer a more functional declaration:

```ruby
def generator_config
  @generator_config ||= Hash[@synapse.available_generators.map do |gen_name, gen|
    gen_opts = opts[gen_name] || {}
    normalized_opts = gen.normalize_config_generator_opts!(gen_opts)
    [gen_name, normalized_opts]
  end]
end
```

this might require us to persist the opts in the initializer:
```ruby
attr_reader :opts
def initialize(opts = {}, synapse)
  @opts = opts
end
```